diff --git a/src/schedulesdbus.cpp b/src/schedulesdbus.cpp
index a9f0f4e..4d05ee9 100755
--- a/src/schedulesdbus.cpp
+++ b/src/schedulesdbus.cpp
@@ -282,8 +282,7 @@ QString CSchedulesDBus::toconvertData(QDateTime date)
 
 QDateTime CSchedulesDBus::fromconvertData(QString str)
 {
-    QStringList liststr = str.split("+", QString::SkipEmptyParts);
-    return QDateTime::fromString(liststr.at(0), "yyyy-MM-ddThh:mm:ss");
+    return QDateTime::fromString(str, Qt::ISODate);
 }
 
 QString CSchedulesDBus::toconvertIGData(QDateTime date)
