diff --color -Naur a/src/customframe.cpp b/src/customframe.cpp
--- a/src/customframe.cpp	2022-02-08 10:03:17.506623669 +0800
+++ b/src/customframe.cpp	2022-02-08 10:04:29.230201125 +0800
@@ -20,6 +20,7 @@
 */
 #include "customframe.h"
 #include <QPainter>
+#include <QPainterPath>
 #include <DPalette>
 #include <QFontMetrics>
 DGUI_USE_NAMESPACE
diff --color -Naur a/src/daymonthview.cpp b/src/daymonthview.cpp
--- a/src/daymonthview.cpp	2022-02-08 10:03:17.506623669 +0800
+++ b/src/daymonthview.cpp	2022-02-08 10:41:36.506917338 +0800
@@ -22,11 +22,14 @@
 #include <QGridLayout>
 #include <QLabel>
 #include <QPainter>
+#include <QPainterPath>
 #include <QEvent>
 #include <QDebug>
 #include <QMessageBox>
+#include <QMouseEvent>
 #include <QTime>
 #include <QQueue>
+#include <QWheelEvent>
 #include "daymonthview.h"
 #include <QSpacerItem>
 #include "constants.h"
diff --color -Naur a/src/draginfographicsview.cpp b/src/draginfographicsview.cpp
--- a/src/draginfographicsview.cpp	2022-02-08 10:03:17.510623644 +0800
+++ b/src/draginfographicsview.cpp	2022-02-08 10:40:00.823335148 +0800
@@ -20,11 +20,18 @@
 */
 #include "draginfographicsview.h"
 
-#include <DMenu>
+#include <QContextMenuEvent>
+#include <QDragEnterEvent>
+#include <QDragMoveEvent>
+#include <QDropEvent>
+#include <QKeyEvent>
+#include <QMenu>
 #include <QMimeData>
+#include <QMouseEvent>
 #include <QJsonParseError>
 #include <QJsonDocument>
 #include <QJsonObject>
+#include <QPainterPath>
 
 #include "schceduledlg.h"
 #include "schcedulectrldlg.h"
diff --color -Naur a/src/schcedulesearchview.cpp b/src/schcedulesearchview.cpp
--- a/src/schcedulesearchview.cpp	2022-02-08 10:03:17.510623644 +0800
+++ b/src/schcedulesearchview.cpp	2022-02-08 10:40:22.971233347 +0800
@@ -19,9 +19,11 @@
 #include "schcedulesearchview.h"
 #include <QAction>
 #include <QMenu>
+#include <QMouseEvent>
 #include <QListWidget>
 #include <QLabel>
 #include <QPainter>
+#include <QPainterPath>
 #include <QHBoxLayout>
 #include <QStylePainter>
 #include <QRect>
diff --color -Naur a/src/weekheadview.cpp b/src/weekheadview.cpp
--- a/src/weekheadview.cpp	2022-02-08 10:03:17.514623618 +0800
+++ b/src/weekheadview.cpp	2022-02-08 10:05:16.469959814 +0800
@@ -22,6 +22,7 @@
 
 #include <QHBoxLayout>
 #include <QPainter>
+#include <QPainterPath>
 #include <QEvent>
 #include <QDebug>
 #include <QMessageBox>
diff --color -Naur a/src/yearwindow.cpp b/src/yearwindow.cpp
--- a/src/yearwindow.cpp	2022-02-08 10:03:17.514623618 +0800
+++ b/src/yearwindow.cpp	2022-02-08 10:42:12.606773558 +0800
@@ -21,9 +21,11 @@
 #include "constants.h"
 #include "calendardbus.h"
 #include <QMessageBox>
+#include <QPainterPath>
 #include <DPalette>
 #include <DHiDPIHelper>
 #include <QMenuBar>
+#include <QWheelEvent>
 #include "schcedulesearchview.h"
 DGUI_USE_NAMESPACE
 CYearWindow::CYearWindow(QWidget *parent): QMainWindow (parent)
